mapper/publication_state.go

```diff
@@ -0,0 +1,62 @@
+package mapper
+
+import (
+	"context"
+	"time"
+
+	"github.com/ONSdigital/dp-frontend-release-calendar/model"
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+func GetPublicationState(description model.ReleaseDescription, dateChanges []model.DateChange) model.PublicationState {
```

Comment:
Public functions should have a go doc
```suggestion
// GetPublicationState returns the publication state
func GetPublicationState(description model.ReleaseDescription, dateChanges []model.DateChange) model.PublicationState {
```